matchingengine/OrderBook_test.go

```diff
@@ -53,3 +67,178 @@ func TestLoadOldAsk(t *testing.T) {
+	// and if it is not initialized, a nil reference error is thrown.
+	config := utils.GetConfiguration()
+	utils.Init(config)
+
+	mockControl := gomock.NewController(t)
+	defer mockControl.Finish()
+
+	var stockID uint32 = 1
+	var stockQuantity uint32 = 10
+	var stockPrice uint32 = 20
+
+	mockAskQueue := mocks.NewMockAskPQueue(mockControl)
+	mockBidQueue := mocks.NewMockBidPQueue(mockControl)
+	mockAskStoplossQueue := mocks.NewMockAskPQueue(mockControl)
+	mockBidStoplossQueue := mocks.NewMockBidPQueue(mockControl)
+	mockDepth := mocks.NewMockMarketDepthStream(mockControl)
+
+	ob := &orderBook{
+		logger: utils.Logger.WithFields(logrus.Fields{
+			"module":        "matchingengine.OrderBook.test",
+			"param_stockId": stockID,
+		}),
+		stockId:     stockID,
+		askChan:     make(chan *models.Ask),
+		bidChan:     make(chan *models.Bid),
+		asks:        mockAskQueue,
+		bids:        mockBidQueue,
+		askStoploss: mockAskStoplossQueue,
+		bidStoploss: mockBidStoplossQueue,
+		depth:       mockDepth,
+	}
+
+	stopLossAsk := makeAsk(1, stockID, models.StopLoss, stockQuantity, stockPrice, "")
+
+	mockAskStoplossQueue.EXPECT().Push(stopLossAsk).Times(1)
+
+	ob.LoadOldAsk(stopLossAsk)
+
+}
+
+func TestLoadOldBid(t *testing.T) {
+
+	// This initializes the logger. This is required because LoadOldBid tries accessing the logger
+	// and if it is not initialized, a nil reference error is thrown.
+	config := utils.GetConfiguration()
+	utils.Init(config)
+
+	mockControl := gomock.NewController(t)
+	defer mockControl.Finish()
+
+	var stockID uint32 = 1
+	var stockQuantity uint32 = 10
+	var stockPrice uint32 = 20
+
+	mockAskQueue := mocks.NewMockAskPQueue(mockControl)
+	mockBidQueue := mocks.NewMockBidPQueue(mockControl)
+	mockAskStoplossQueue := mocks.NewMockAskPQueue(mockControl)
+	mockBidStoplossQueue := mocks.NewMockBidPQueue(mockControl)
+	mockDepth := mocks.NewMockMarketDepthStream(mockControl)
+
+	ob := &orderBook{
+		logger: utils.Logger.WithFields(logrus.Fields{
+			"module":        "matchingengine.OrderBook.test",
+			"param_stockId": stockID,
+		}),
+		stockId:     stockID,
+		askChan:     make(chan *models.Ask),
+		bidChan:     make(chan *models.Bid),
+		asks:        mockAskQueue,
+		bids:        mockBidQueue,
+		askStoploss: mockAskStoplossQueue,
+		bidStoploss: mockBidStoplossQueue,
+		depth:       mockDepth,
+	}
+
+	limitBid := makeBid(1, stockID, models.Limit, stockQuantity, stockPrice, "")
+
+	mockBidQueue.EXPECT().Push(limitBid).Times(1)
+	mockDepth.EXPECT().AddOrder(false, false, stockPrice, stockQuantity).Times(1)
+
+	ob.LoadOldBid(limitBid)
+
+}
+
+func TestLoadOldBidStopLoss(t *testing.T) {
+
+	// This initializes the logger. This is required because LoadOldAsk tries accessing the logger
+	// and if it is not initialized, a nil reference error is thrown.
+	config := utils.GetConfiguration()
+	utils.Init(config)
+
+	mockControl := gomock.NewController(t)
+	defer mockControl.Finish()
+
+	var stockID uint32 = 1
```

Comment:
```stockId := 1```

Reply:
that will make stockId as an `int`. I used this to cast it to `uint32` which is what it needs to be.

Reply:
Won't ```stockId := uint32(1)``` work? I mean I really don't care about using ```var``` or ```:=``` notation but just putting it there :stuck_out_tongue: